src/Persistence/Sql/Sqlite/CreateRegexpReplaceFunctionMiddleware.php

```diff
@@ -30,18 +30,14 @@ public function connect(
                         return null;
                     }
 
-                    if (is_int($value)) {
-                        $value = (string) $value;
-                    } elseif (is_float($value)) {
-                        $value = Expression::castFloatToString($value);
-                    }
+                    $value = CreateRegexpLikeFunctionMiddleware::castScalarToString($value);
 
-                    $isValidUtf8Value = \PHP_VERSION_ID < 80200
-                        ? preg_match('~~u', $value) === 1 // much faster in PHP 8.1 and lower
```

Comment:
is this performance flaw fixed in newer PHP 8.2, 8.3 versions?

Reply:
discussed in https://github.com/atk4/data/pull/1202#discussion_r1578233748